examples/ex27.cpp

```diff
@@ -0,0 +1,718 @@
+
+      // Counteract the n.Grad(u) term on the Dirichlet portion of the boundary
+      a.AddBdrFaceIntegrator(new DGDiffusionIntegrator(matCoef, sigma, kappa),
+                             dbc_bdr);
+
+      // Augment the n.Grad(u) term with a*u on the Robin portion of boundary
+      a.AddBdrFaceIntegrator(new BoundaryMassIntegrator(m_rbcACoef),
+                             rbc_bdr);
+   }
+   a.Assemble();
+
+   // 8. Assemble the linear form for the right hand side vector.
+   LinearForm b(&fespace);
+
+   if (h1)
+   {
+      // Set the Dirchlet values in the solution vector
+      u.ProjectBdrCoefficient(dbcCoef, dbc_bdr);
+
+      // Add the desired value for n.Grad(u) on the Neumann boundary
+      b.AddBoundaryIntegrator(new BoundaryLFIntegrator(m_nbcCoef), nbc_bdr);
+
+      // Add the desired value for n.Grad(u) + a*u on the Robin boundary
+      b.AddBoundaryIntegrator(new BoundaryLFIntegrator(m_rbcBCoef), rbc_bdr);
+   }
+   else
+   {
+      // Add the desired value for the Dirchlet boundary
+      b.AddBdrFaceIntegrator(new DGDirichletLFIntegrator(dbcCoef, matCoef,
+                                                         sigma, kappa),
+                             dbc_bdr);
+
+      // Add the desired value for n.Grad(u) on the Neumann boundary
+      b.AddBdrFaceIntegrator(new BoundaryLFIntegrator(m_nbcCoef),
+                             nbc_bdr);
+
+      // Add the desired value for n.Grad(u) + a*u on the Robin boundary
+      b.AddBdrFaceIntegrator(new BoundaryLFIntegrator(m_rbcBCoef),
+                             rbc_bdr);
+   }
+   b.Assemble();
+
+   // 9. Construct the linear system.
+   OperatorPtr A;
+   Vector B, X;
+   a.FormLinearSystem(ess_tdof_list, u, b, A, X, B);
+
+#ifndef MFEM_USE_SUITESPARSE
+   // 10. Define a simple symmetric Gauss-Seidel preconditioner and use it to
+   //     solve the system AX=B with PCG in the symmetric case, and GMRES in the
+   //     non-symmetric one.
+   {
+      GSSmoother M((SparseMatrix&)(*A));
+      if (sigma == -1.0)
+      {
+         PCG(*A, M, B, X, 1, 500, 1e-12, 0.0);
+      }
+      else
+      {
+         GMRES(*A, M, B, X, 1, 500, 10, 1e-12, 0.0);
+      }
+   }
+#else
+   // 11. If MFEM was compiled with SuiteSparse, use UMFPACK to solve the
+   //     system.
+   UMFPackSolver umf_solver;
+   umf_solver.Control[UMFPACK_ORDERING] = UMFPACK_ORDERING_METIS;
+   umf_solver.SetOperator(*A);
+   umf_solver.Mult(B, X);
+#endif
+
+   // 12. Recover the grid function corresponding to U. This is the
+   //     local finite element solution.
+   a.RecoverFEMSolution(X, b, u);
+
+   // 13. Build a mass matrix to help solve for n.Grad(u) where 'n' is
+   //     a surface normal.
+   BilinearForm m(&fespace);
+   m.AddDomainIntegrator(new MassIntegrator);
+   m.Assemble();
+
+   ess_tdof_list.SetSize(0);
+   OperatorPtr M;
+   m.FormSystemMatrix(ess_tdof_list, M);
+
+   // 14. Compute the various boundary integrals.
```

Comment:
It may be good to clarify here that we only compute the integral (or is it the average?) of the errors on the respective parts of the boundary. In particular, these are not norms of the error and therefore may be zero (or very small) while the error is not zero (or small). Even with constant exact b.c. these integrals (or averages) may be misleading.

Reply:
You make a very good point.  My original intention was to compute the l2 norm of the BC values on the boundary and compare this to the expected result.  For the H1 case this worked fine but not the for DG case.  The problem was that we cannot evaluate `GridFunctionCoefficients` based on DG `GridFunctions` on boundaries until something like PR #1194 is in place.

I agree that we don't want to be misleading so I'll correct the integrals so that they compute the l2 norm in the H1 case and I'll put a work-around in place to handle the DG case.

Reply:
Actually, I couldn't get the l2 norm to work for the H1 case either.  However, I was able to implement an l1 error measure (if I'm not mistaken).